org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPErrorReporterForXML.java

```diff
@@ -49,15 +48,17 @@
 
 	private Set<ReferencedGrammarInfo> referencedGrammars;
 
-	private Map<String, ReferencedGrammarDiagnosticsInfo> referencedGrammarDiagnosticsInfoCache;
+	private final Map<String, ReferencedGrammarDiagnosticsInfo> referencedGrammarDiagnosticsInfoCache;
 
 	private final boolean hasRelatedInformation;
 
 	public LSPErrorReporterForXML(DOMDocument xmlDocument, List<Diagnostic> diagnostics,
-			ContentModelManager contentModelManager, boolean hasRelatedInformation) {
+			ContentModelManager contentModelManager, boolean hasRelatedInformation,
+			Map<String, ReferencedGrammarDiagnosticsInfo> referencedGrammarDiagnosticsInfoCache) {
 		super(XML_DIAGNOSTIC_SOURCE, xmlDocument, diagnostics);
 		this.contentModelManager = contentModelManager;
 		this.hasRelatedInformation = hasRelatedInformation;
+		this.referencedGrammarDiagnosticsInfoCache = referencedGrammarDiagnosticsInfoCache;

```

Comment:
referencedGrammarDiagnosticsInfoCache ==null? new HashMap():referencedGrammarDiagnosticsInfoCache

to prevent NPEs down the line and unnecessary null checks

Reply:
fixed